plex_generate_previews.py

```diff
@@ -8,23 +9,84 @@
 import glob
 import os
 import struct
-import gpustat
+
+if not os.path.isfile('/usr/bin/mediainfo'):
+    print('MediaInfo not found.  MediaInfo must be installed and available in PATH.')
+    sys.exit()	
+
+try:
+    from pymediainfo import MediaInfo
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install pymediainfo".')
+    sys.exit()
+
+try:
+    import gpustat
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install gpustat".')
+    sys.exit()
+ 
 import time
-import requests
+
+try:
+    import requests
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install requests".')
+    sys.exit()
+
 import array
-from plexapi.server import PlexServer
-from loguru import logger
-from rich.console import Console
-from rich.progress import Progress, SpinnerColumn, MofNCompleteColumn
+
+try:
+    from plexapi.server import PlexServer
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install plexapi".')
+    sys.exit()
+    
+try:
+    from loguru import logger
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install loguru".')
+    sys.exit()
+
+try:
+    from rich.console import Console
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install rich".')
+    sys.exit()
+
+try:
+    from rich.progress import Progress, SpinnerColumn, MofNCompleteColumn
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install rich".')
+    sys.exit()
+
+try:
+    from PIL import Image
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install pillow".')
+    sys.exit()
+
+import PIL
+
+if not os.path.isfile('/usr/bin/ffmpeg'):
+    print('FFmpeg not found.  FFmpeg must be installed and available in PATH.')
+    sys.exit()	
 
 # EDIT These Vars #
 PLEX_URL = 'https://xxxxxx.plex.direct:32400/'
```

Comment:
@jasonsansone thoughts on moving these vars to the top of the file as they are now a fair way down due to the import statements try/catch? Or we could make a common function for the import statement to simplify.